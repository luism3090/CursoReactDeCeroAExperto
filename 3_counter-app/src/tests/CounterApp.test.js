import React, { Children } from 'react'
const { shallow } = require("enzyme");
const { default: CounterApp } = require("../CounterApp");

describe('pruebas en el <CounterApp >', () => {
    
    test('Debe mostrar <Counter App> correctamente', () => {
        
        const wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar <Counter App> value = 100', () => {
        
        const wrapper = shallow(<CounterApp value={100}/>);

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('100');
        

    });


    test('Debe validar el incremento del boton 1+', () => {
        
        const wrapper = shallow(<CounterApp />);

        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('11');

    });

    test('Debe validar el decremento del boton 1-', () => {
        
        const wrapper = shallow(<CounterApp />);

        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('9');

    });

    test('Debe colocar el valor por defecto de 10', () => {
        
        const wrapper = shallow(<CounterApp value={105}/>);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');

        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        //console.log(counterText);

        expect(counterText).toBe('105');

    })
    
    
    


});
