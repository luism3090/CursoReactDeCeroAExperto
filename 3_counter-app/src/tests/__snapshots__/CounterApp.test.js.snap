// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <CounterApp > Debe mostrar <Counter App> correctamente 1`] = `
<div>
  <h3>
    ----------------- Counter App ---------------
  </h3>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    1+
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    1-
  </button>
</div>
`;
